lib/workers/repository/process/fetch.ts

```diff
@@ -102,15 +102,13 @@ async function fetchManagerPackagerFileUpdates(
   pFile: PackageFile
 ): Promise<void> {
   const { packageFile } = pFile;
+  const packageFileConfig = mergeChildConfig(managerConfig, pFile);
   if (pFile.extractedConstraints) {
-    pFile.constraints = {
+    packageFileConfig.constraints = {
       ...pFile.extractedConstraints,
       ...config.constraints,
-      ...pFile.constraints,
     };
-    delete pFile.extractedConstraints;
```

Comment:
Ideally we could still delete here and artifacts routines would "re-extract", but that's a change for another day